src/Microsoft.VisualStudio.Threading/JoinableTaskContext.cs

```diff
@@ -116,6 +116,11 @@ public partial class JoinableTaskContext : IDisposable
         /// </summary>
         private readonly int mainThreadManagedThreadId;
 
+        /// <summary>
+        /// The count of <see cref="JoinableTaskFactory"/> blocking API calls on the main thread.
+        /// </summary>
+        private int mainThreadJTFBlockingCount;
```

Comment:
Since this isn't accessed with locks, and it can be read from other threads than the main thread that writes to it, I believe we should use the `volatile` modifier on this field so that we never read a value from a CPU core's local cache that may be stale.

Reply:
Updated